components/contribution-flow/StepDetailsCrypto.js

```diff
@@ -26,14 +27,57 @@ const messages = defineMessages({
   },
 });
 
-const StepDetailsCrypto = ({ onChange, data }) => {
+/*
+ * Calculates the approximate value for 1 unit of a given crypto currency in the collective currency.
+ * Uses the Cryptonator API (https://www.cryptonator.com/api/)
+ */
+const getCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+  if (!cryptoCurrency || !collectiveCurrency) {
+    return null;
+  }
+  try {
+    const response = await fetch(`https://api.cryptonator.com/api/ticker/${cryptoCurrency}-${collectiveCurrency}`);
+    const body = await response.json();
+    if (!body.success || body.error !== '') {
+      throw new Error(`Cryptonator Error: ${body.error}`);
+    }
+    return body.ticker.price;
+  } catch (error) {
+    // we don't want the user to see any errors; simply don't show the conversion amount
+    // eslint-disable-next-line no-console
+    console.error(error);
+    return null;
+  }
+};
+
+const StepDetailsCrypto = ({ onChange, data, currency }) => {
   const intl = useIntl();
-  const [currencyType, setCurrencyType] = useState(data.currency);
+  const [cryptoCurrencyType, setCryptoCurrencyType] = useState(data.currency);
   const [amount, setAmount] = useState(data.amount);
+  const [convertedAmount, setConvertedAmount] = useState(data.convertedAmount?.amount);
```

Comment:
What is this? Where are these values coming from? `data.currency`, `data.amount`, `data.convertedAmount?.amount`

Reply:
These are coming from the `stepDetails`; 

https://github.com/opencollective/opencollective-frontend/blob/5d9417688f0913a9690d27d6514040f6b75f5146/components/contribution-flow/ContributionFlowStepContainer.js#L140

We pass them around using the `dispatchChange` function so that they are available to other steps in the contribution flow. However I realized that we don't need to pass the converted amount (as per the [above comment](https://github.com/opencollective/opencollective-frontend/pull/6787#discussion_r700347283)) and thus I've set it to null.